packages/elements-core/src/components/TryIt/MockingButton.tsx

```diff
@@ -23,88 +22,95 @@ export const MockingButton: React.FC<MockingButtonProps> = ({
 
   const operationResponses = operation.responses;
 
-  const setMockingOptions = ({ code, example, dynamic }: Omit<MockingOptions, 'isEnabled'>) => {
-    onOptionsChange({ isEnabled, code, example, dynamic });
-  };
+  const setMockingOptions = React.useCallback(
+    ({ code, example, dynamic }: Omit<MockingOptions, 'isEnabled'>) => {
+      onOptionsChange({ isEnabled, code, example, dynamic });
+    },
+    [isEnabled, onOptionsChange],
+  );
+
+  const menuItems = React.useMemo(() => {
+    const items: MenuItems = [
+      { id: 'mocking-enabled', title: 'Enabled', isChecked: isEnabled, onPress: toggleEnabled },
+      {
+        type: 'group',
+        children: operationResponses
+          ?.filter(operationResponse => Number.isInteger(parseFloat(operationResponse.code)))
+          ?.map(generateOperationResponseMenu),
+      },
+    ];
+
+    function generateOperationResponseMenu(operationResponse: IHttpOperationResponse) {
+      const menuId = `response-${operationResponse.code}`;
+      const isActive = operationResponse.code === code;
+      const exampleKeys = uniq(operationResponse.contents?.flatMap(c => c.examples || []).map(example => example.key));
+
+      const exampleChildren: MenuActionItem[] = exampleKeys?.map(exampleKey => ({
+        id: `${menuId}-example-${exampleKey}`,
+        title: exampleKey,
+        isChecked: isActive && exampleKey === example,
+        onPress: () => {
+          setMockingOptions({ code: operationResponse.code, example: exampleKey });
+        },
+      }));
+
+      const generationModeItems: MenuActionItem[] = [
+        {
+          id: `${menuId}-gen-static`,
+          title: 'Statically Generated',
+          isChecked: isActive && dynamic === false,
+          onPress: () => {
+            setMockingOptions({ code: operationResponse.code, dynamic: false });
+          },
+        },
+        {
+          id: `${menuId}-gen-dynamic`,
+          title: 'Dynamically Generated',
+          isChecked: isActive && dynamic === true,
+          onPress: () => {
+            setMockingOptions({ code: operationResponse.code, dynamic: true });
+          },
+        },
+      ];
+
+      const menuItem: MenuItemWithSubmenu = {
+        id: menuId,
+        isDisabled: !isEnabled,
+        isChecked: isActive,
+        title: operationResponse.code,
+        onPress: () => {
+          setMockingOptions({ code: operationResponse.code, dynamic: false });
+        },
+        children: [
+          { type: 'group', children: generationModeItems },
+          { type: 'group', title: 'Examples', children: exampleChildren },
+        ],
+      };
+
+      return menuItem;
+    }
+
+    return items;
+  }, [code, dynamic, example, isEnabled, operationResponses, setMockingOptions, toggleEnabled]);
 
   return (
     <Box>
```

Comment:
Here's what it looks like in practice with the updated menu.

![2021-06-14 16 11 19](https://user-images.githubusercontent.com/847542/121953365-13858900-cd23-11eb-9d71-08b9aa8266cc.gif)